lib/ansible/modules/web_infrastructure/ansible_tower/tower_group.py

```diff
@@ -113,47 +126,74 @@ def main():
         group_by=dict(),
         source_script=dict(),
-        overwrite=dict(type='bool', default=False),
+        overwrite=dict(type='bool'),
         overwrite_vars=dict(),
-        update_on_launch=dict(type='bool', default=False),
+        update_on_launch=dict(type='bool'),
+        merge_variables=dict(required=False, default=False, type='bool'),
+        parent_groups=dict(required=False, default=[], type='list'),
         state=dict(choices=['present', 'absent'], default='present'),
     )
 
     module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
 
+    if FAILED_TOWER_IMPORT:
+        module.fail_json(msg="Failed to import tower_cli. Try installing via Pip using 'pip install ansible-tower-cli'")
+
     name = module.params.get('name')
     inventory = module.params.get('inventory')
     credential = module.params.get('credential')
     state = module.params.pop('state')
+    merge_variables = module.params.pop('merge_variables')
+    parent_groups = module.params.pop('parent_groups')
 
     variables = module.params.get('variables')
-    if variables:
-        if variables.startswith('@'):
-            filename = os.path.expanduser(variables[1:])
-            with open(filename, 'r') as f:
-                variables = f.read()
 
     json_output = {'group': name, 'state': state}
 
     tower_auth = tower_auth_config(module)
     with settings.runtime_values(**tower_auth):
         tower_check_mode(module)
         group = tower_cli.get_resource('group')
-        try:
-            params = module.params.copy()
-            params['create_on_missing'] = True
-            params['variables'] = variables
+        params = module.params.copy()
+        params['create_on_missing'] = True
 
+        try:
             inv_res = tower_cli.get_resource('inventory')
             inv = inv_res.get(name=inventory)
             params['inventory'] = inv['id']
+        except (exc.NotFound) as excinfo:
+            module.fail_json(msg='Failed to update the group, inventory not found: {0}'.format(excinfo), changed=False)
 
+        parent_group_ids = []
+        try:
+            for p_group in parent_groups:
+                parent_group_ids.append(group.get(name=p_group, inventory=params['inventory'])['id'])
+        except exc.NotFound as excinfo:
+            module.fail_json(msg='Parent group {0} does not exist: {1}'.format(p_group, excinfo),
+                             changed=False)
+
+        try:
+            existing_group = group.get(name=name, inventory=inv['id'])
+        except exc.NotFound:
+            existing_group = None
+            json_output['created'] = True
+            module.log("Existing group not found, will create")
+
+        try:
+            if merge_variables and existing_group:
+                params['variables'] = sanitise_and_merge_variables(existing_group['variables'], variables)
+            else:
+                params['variables'] = sanitise_and_merge_variables(variables)
```

Comment:
if `variables` argument is not provided, then I think it's safest to just not put it in `params`. This is also relevant to most uses of `state: absent`

Reply:
> * merge_variables causes me some pause, I think it might be easiest to punt that to do later, there are some duplicate efforts in tower-cli, and there are a couple of nit-picky things to deal with. I would be fine with it here, as long as I was confident it won't affect cases that aren't using the parameter.

I can try and split the changes for merge variables into a 2nd branch/PR so it can be further discussed while allowing the parent group and host group associations to proceed